stacks/portstack.go

```diff
@@ -376,29 +376,41 @@ func (ps *PortStack) HandleEth(dst []byte) (n int, err error) {
 //
 // If a handler returns any other error the port is closed.
 func (ps *PortStack) handleEth(dst []byte) (n int, err error) {
+	//this would be much better named as getNextOutboundPacket
+
+	//println ("portStack:handleEth ",string(dst)) //@@@REMOVE
+
 	switch {
 	case len(dst) < int(ps.mtu):
+		
 		return 0, io.ErrShortBuffer
 
 	case !ps.IsPendingHandling():
+		
 		return 0, nil // No remaining packets to handle.
 	}
 	n = ps.arpClient.handle(dst)
 	if n != 0 {
 		return n, nil
 	}
 
+	//i don't know why this is defined within the scope of handleEth
 	type Socket interface {
 		Close()
 		IsPendingHandling() bool
-		HandleEth(dst []byte) (int, error)
+		 HandleEth(dst []byte) (int, error)
 	}
-
+	//or why this is an anonymous function is stored in a local variable before invocation  - I can see it narrows scope, but it's an odd pattern and confusing as the code does not run in the order it appears
 	handleSocket := func(dst []byte, sock Socket) (int, bool, error) {
+		
+		//println ("handleSocket called") //@@@REMOVE
+		//note sock is a UDPport or TCPport - things that impliment the Sock interface
+		
```

Comment:
remove comments, the second comment is OK but belongs above the Socket type definition.